code/datums/ai/_ai_controller.dm

```diff
@@ -7,6 +7,23 @@ multiple modular subtrees with behaviors
 /datum/ai_controller
 	///The atom this controller is controlling
 	var/atom/pawn
+	/**
+	 * This is a list of variables the AI uses and can be mutated by actions.
+	 *
+	 * When an action is performed you pass this list and any relevant keys for the variables it can mutate.
+	 *
+	 * DO NOT set values in the blackboard directly, and especially not if you're adding a datum reference to this!
+	 * Use the setters, this is important for reference handing.
+	 *
+	 * Does not (currently) support three-level deep lists.
```

Comment:
why not?

Reply:
you're talking about `sig_remove_from_blackboard` right?

Reply:
![cat](https://user-images.githubusercontent.com/58055496/232997932-47a1a1aa-77d6-45ec-812e-9afcd8abad31.jpg)